src/buffer/out/textBuffer.cpp

```diff
@@ -1395,17 +1406,19 @@ bool TextBuffer::MoveToPreviousWord(COORD& pos, std::wstring_view wordDelimiters
 // - pos - a COORD on the word you are currently on
 // Return Value:
 // - pos - The COORD for the first cell of the current glyph (inclusive)
-const til::point TextBuffer::GetGlyphStart(const til::point pos) const
+const til::point TextBuffer::GetGlyphStart(const til::point pos, std::optional<til::point> limitOptional) const
 {
     COORD resultPos = pos;
     const auto bufferSize = GetSize();
+    const auto limit{ limitOptional ? *limitOptional : bufferSize.EndExclusive() };
 
-    if (resultPos == bufferSize.EndExclusive())
+    // Clamp pos to limit
+    if (bufferSize.CompareInBounds(resultPos, limit, true) > 0)
     {
-        bufferSize.DecrementInBounds(resultPos, true);
+        resultPos = limit;
     }
 
-    if (resultPos != bufferSize.EndExclusive() && GetCellDataAt(resultPos)->DbcsAttr().IsTrailing())
+    if (resultPos != limit && GetCellDataAt(resultPos)->DbcsAttr().IsTrailing())

```

Comment:
Comment explaining why? I know you just changed the limit... but we should say like "We can't sit on the right half of a 2 column wide character so move to the left."